src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CodeSnippets.cs

```diff
@@ -635,37 +635,121 @@ partial class Test
         [MarshalUsing(typeof({nativeTypeName}))] out {typeName} pOut);
 }}
 ";
-        public static string BasicNonBlittableUserDefinedType(bool defineNativeMarshalling = true) => $@"
+
+        public static string BasicParameterWithByRefModifier(string byRefKind, string typeName, string preDeclaration = "") => $@"
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+{preDeclaration}
+partial class Test
+{{
+    [LibraryImport(""DoesNotExist"")]
+    public static partial void Method(
+        {byRefKind} {typeName} p);
+}}";
+
+        public static string BasicParameterByValue(string typeName, string preDeclaration = "") => $@"
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+{preDeclaration}
+partial class Test
+{{
+    [LibraryImport(""DoesNotExist"")]
+    public static partial void Method(
+        {typeName} p);
+}}";
+
+        public static string BasicReturnType(string typeName, string preDeclaration = "") => $@"
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+{preDeclaration}
+partial class Test
+{{
+    [LibraryImport(""DoesNotExist"")]
+    public static partial {typeName} Method();
+}}";
+
+        public static string BasicReturnAndParameterByValue(string returnType, string parameterType, string preDeclaration = "") => $@"
+using System.Runtime.InteropServices;
+{preDeclaration}
+partial class Test
+{{
+    [LibraryImport(""DoesNotExist"")]
+    public static partial {returnType} Method({parameterType} p);
+}}";
+
+        public static class CustomStructMarshalling
+        {
+            public static string NonBlittableUserDefinedType(bool defineNativeMarshalling = true) => @$"
 {(defineNativeMarshalling ? "[NativeMarshalling(typeof(Native))]" : string.Empty)}
+[StructLayout(LayoutKind.Sequential)]
```

Comment:
This is the default for `struct`s. Do we need to add this for some reason?

Reply:
This was in the existing snippets - let me try removing it.

Reply:
Looks like this we get `CS0649` without it... We can probably add that to our list of allowed diagnostics in `AssertPreSourceGeneratorCompilation`.